### ISSUE TITLE
Distribute a Flatpak

### ISSUE THREAD

**Issue by PopeRigby** [beyond-all-reason/Beyond-All-Reason#119]
It makes it easier to get the game on any distribution and keep it up-to-date.

**Comment by lhog** [comment]
Can you summarize why the current AppImage executable is not good?

**Comment by gajop** [comment]
It's not really an option as the build system that we use doesn't support it.
I think only AppImage and snap are an option. As a Linux users I don't have any issues with AppImage.. but maybe I'm missing something..

See more here:  https://www.electron.build/configuration/linux

**Comment by gajop** [comment]
Still an open issue, since 2016: https://github.com/electron-userland/electron-builder/issues/512

**Comment by PopeRigby** [comment]
> Can you summarize why the current AppImage executable is not good?

Mostly because it makes it hard to keep it up-to-date. If I want to update the AppImage, I have to download it again and replace the old one. With Flatpak, I just run `flatpak update` and all my Flatpaks are up-to-date. Also, the system integration isn't as good. Flatpak handles creating a `.desktop` file, but integrating AppImages requires fiddling with `appimaged`, which I was never able to get working.

**Comment by gajop** [comment]
> If I want to update the AppImage, I have to download it again and replace the old one.

It'll actually update itself on start, as a user you don't really have to manually update it. I personally launch most things on Linux via the command line so I don't have any issues with desktop integration, so your concerns are probably valid.

What's your opinion on snap? It seems very well integrated in Ubuntu. As I said before, flatpak isn't really an option for us as the distribution system we use doesn't support it.

**Comment by PopeRigby** [comment]
> What's your opinion on snap? It seems very well integrated in Ubuntu. As I said before, flatpak isn't really an option for us as the distribution system we use doesn't support it.

I'm not the biggest fan, but I don't have a ton of experience with them. Maybe take this into consideration, because it seems to do a good job of going over why Snap has generated controversy: https://www.zdnet.com/article/linux-mint-dumps-ubuntu-snap/

**Comment by gajop** [comment]
Thanks for the article. 
As an Ubuntu user, snap would work for me as I already have a lot of snap applications but I can see how this would be a bad option for other Linux users.

With that in mind I think I'll keep it as AppImage for the time being. Maybe in the future there will be .deb or .rpm builds but I don't see a move to snap or flatpak  happening in the near future.

**Comment by tobias47n9e** [comment]
@lumbo7332 The Flatpak is now here: https://flathub.org/apps/details/info.beyondallreason.bar
